pkg/operator/configobservation/images/observe_images.go

```diff
@@ -1,42 +1,111 @@
+
+	"github.com/openshift/cluster-openshift-apiserver-operator/pkg/operator/configobservation"
 )
 
-func ObserveInternalRegistryHostname(genericListers configobserver.Listers, existingConfig map[string]interface{}) (map[string]interface{}, []error) {
+func ObserveImageConfig(genericListers configobserver.Listers, existingConfig map[string]interface{}) (map[string]interface{}, []error) {
 	listers := genericListers.(configobservation.Listers)
 	var errs []error
 	prevObservedConfig := map[string]interface{}{}
 
+	// first observe all the existing config values so that if we get any errors
+	// we can at least return those.
 	internalRegistryHostnamePath := []string{"imagePolicyConfig", "internalRegistryHostname"}
 	if currentInternalRegistryHostname, _, _ := unstructured.NestedString(existingConfig, internalRegistryHostnamePath...); len(currentInternalRegistryHostname) > 0 {
-		unstructured.SetNestedField(prevObservedConfig, currentInternalRegistryHostname, internalRegistryHostnamePath...)
+		err := unstructured.SetNestedField(prevObservedConfig, currentInternalRegistryHostname, internalRegistryHostnamePath...)
+		if err != nil {
+			return prevObservedConfig, append(errs, err)
+		}
+	}
+
+	externalRegistryHostnamePath := []string{"imagePolicyConfig", "externalRegistryHostnames"}
+	o, _, err := unstructured.NestedSlice(existingConfig, externalRegistryHostnamePath...)
+	err = unstructured.SetNestedSlice(prevObservedConfig, o, externalRegistryHostnamePath...)
+	if err != nil {
+		return prevObservedConfig, append(errs, err)
+	}
+
+	allowedRegistriesForImportPath := []string{"imagePolicyConfig", "allowedRegistriesForImport"}
+	o, _, err = unstructured.NestedSlice(existingConfig, allowedRegistriesForImportPath...)
+	err = unstructured.SetNestedSlice(prevObservedConfig, o, allowedRegistriesForImportPath...)
+	if err != nil {
+		return prevObservedConfig, append(errs, err)
 	}
 
 	if !listers.ImageConfigSynced() {
 		glog.Warning("images.config.openshift.io not synced")
-		return prevObservedConfig, errs
+		return prevObservedConfig, append(errs, err)
 	}
 
+	// now gather the cluster config and turn it into the observed config
 	observedConfig := map[string]interface{}{}
 	configImage, err := listers.ImageConfigLister.Get("cluster")
 	if errors.IsNotFound(err) {
 		glog.Warningf("image.config.openshift.io/cluster: not found")
-		return observedConfig, errs
+		return observedConfig, append(errs, err)
 	}
 	if err != nil {
-		return prevObservedConfig, errs
+		return prevObservedConfig, append(errs, err)
 	}
+
 	internalRegistryHostName := configImage.Status.InternalRegistryHostname
 	if len(internalRegistryHostName) > 0 {
-		unstructured.SetNestedField(observedConfig, internalRegistryHostName, internalRegistryHostnamePath...)
+		err = unstructured.SetNestedField(observedConfig, internalRegistryHostName, internalRegistryHostnamePath...)
+		if err != nil {
+			return observedConfig, append(errs, err)
```

Comment:
@deads2k @sanchezl it's not obvious to me whether errors at this point in the process should return prevObservedConfig or observedConfig.  Returning observedConfig here means we're potentially throwing away existing config.  Returning prevObservedConfig means potentially returning invalid/stale information.

(same applies to all the other places where we catch an error and return observedConfig below)